src/OrbitGl/GpuTrack.cpp

```diff
@@ -111,14 +111,14 @@ float GpuTrack::GetHeight() const {
   if (IsCollapsed()) {
     return submission_track_->GetHeight();
   }
-  float height = layout_->GetTrackTabHeight();
+  float height = 0;
   if (submission_track_->ShouldBeRendered()) {
-    height += submission_track_->GetHeight();
-    height += layout_->GetSpaceBetweenSubtracks();
+    float additional_height = submission_track_->GetHeight() + layout_->GetSpaceBetweenSubtracks();
+    height += std::max(additional_height, 2.f * layout_->GetThreadTrackMinimumHeight());
```

Comment:
Still feels a bit like a hack, but I guess its fine for now.

Reply:
I'm not sure this is the right place to enforce the constraint, and I wouldn't re-use the `GetThreadTrackMinimumHeight()` because the multiplication with 2.f seems a bit random here.

At least introduce a separate layout variable (e.g. `SubTrackMinimumHeight`), then I'm fine with this here. Alternatively, the correct way could be enforcing this in the child tracks - i.e. the submission track should not report a height smaller than allowed. This was my first instinct, but then they should only do that *if* they are actually subtracks (i.e. indentation > 0). I don't have a strong opinion either way.

Reply:
Added a comment, this will go away very soon when we implement a new track hierarchy system.